src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -168,87 +110,44 @@ allocate(const std::vector<int>& inteHead)
-    if ((childGroups.size() != 0) && (childWells.size()!=0))
-        throw std::invalid_argument("group has both wells and child groups" + group.name());
-    int igrpCount = 0;
-    if (childWells.size() != 0) {
+    if (group.wellgroup()) {
+        int igrpCount = 0;
         //group has child wells
         //store the well number (sequence index) in iGrp according to the sequence they are defined
-        for ( const auto& well : childWells) {
-            iGrp[igrpCount] = well.seqIndex()+1;
-            igrpCount+=1;
+        for (const auto& well_name : group.wells()) {
+            const auto& well = sched.getWell2(well_name, simStep);
+            iGrp[igrpCount] = well.seqIndex() + 1;
+            printf("setter IGRP[%d] = %ld (%s)\n", igrpCount, well.seqIndex() + 1, well.name().c_str());
+            igrpCount += 1;
         }
-    }
-    else if (childGroups.size() != 0) {
-        //group has child groups
-        //The field group always has seqIndex = 0 because it is always defined first
-        //Hence the all groups except the Field group uses the seqIndex assigned
-        //iGrp contains the child groups in ascending group sequence index
-        std::vector<size_t> childGroupIndex;
-        Opm::RestartIO::Helpers::groupMaps groupMap;
-        groupMap.currentGrpTreeNameSeqIndMap(sched, simStep, groupMapNameIndex,mapIndexGroup);
-        const auto indGroupMap = groupMap.indexGroupMap();
-        const auto gNameIndMap = groupMap.groupNameIndexMap();
-        for (auto* grp : childGroups) {
-            auto groupName = grp->name();
-            auto searchGTName = gNameIndMap.find(groupName);
-            if (searchGTName != gNameIndMap.end()) {
-                childGroupIndex.push_back(searchGTName->second);
-            }
-            else {
-                throw std::invalid_argument( "Invalid group name" );
-            }
-        }
-        std::sort(childGroupIndex.begin(), childGroupIndex.end());
-        for (auto groupTreeIndex : childGroupIndex) {
-            auto searchGTIterator = indGroupMap.find(groupTreeIndex);
-            if (searchGTIterator != indGroupMap.end()) {
-                auto gname = (searchGTIterator->second)->name();
-                auto gSeqNoItr = groupMapNameIndex.find(gname);
-                if (gSeqNoItr != groupMapNameIndex.end()) {
-                    iGrp[igrpCount] = (gSeqNoItr->second) + 1;
-                    igrpCount+=1;
-                }
-                else {
-                    std::cout << "AggregateGroupData, ChildGroups - Group name: groupMapNameIndex: " << gSeqNoItr->first << std::endl;
-                    throw std::invalid_argument( "Invalid group name" );
-                }
-            }
-            else {
-                std::cout << "AggregateGroupData, ChildGroups - GroupTreeIndex: " << groupTreeIndex << std::endl;
-                throw std::invalid_argument( "Invalid GroupTree index" );
-            }
+    } else if (group.groups().size() != 0) {
```

Comment:
Function `Group2::groups()` returns a `std::vector<>`.  This is more idiomatically expressed as
```C++
! group.groups().empty()
```